syntax = "proto3";

package node;

service NodeService {
  rpc Ping (PingRequest) returns (PingResponse);
  rpc GetNodeStatus (StatusRequest) returns (StatusResponse);
}

message PingRequest {
  string message = 1;
}

message PingResponse {
  string reply = 1;
}

message StatusRequest {}

message StatusResponse {
  string node_id = 1;
  int32 total_votes = 2;
  string status = 3;
}